Sources/Basics/ArrayHelpers.swift

```diff
@@ -17,5 +17,57 @@ public func nextItem<T: Equatable>(in array: [T], after item: T) -> T? {
             return nextIndex < array.count ? array[nextIndex] : nil
         }
     }
-    return nil // Item not found or it's the last item
+    return nil  // Item not found or it's the last item
+}
+
+/// Determines if an array contains all elements of a subset array in any order.
+/// - Parameters:
+///   - array: The array to search within
+///   - subset: The subset array to check for
+///   - shouldBeContiguous: if `true`, the subset match must be contiguous sequenence
+/// - Returns: `true` if all elements in `subset` are present in `array`, `false` otherwise
+public func contains<T: Equatable>(array: [T], subset: [T], shouldBeContiguous: Bool = true) -> Bool {
+    if shouldBeContiguous {
+        return containsContiguousSubset(array: array, subset: subset)
+    } else {
+        return containsNonContiguousSubset(array: array, subset: subset)
+    }
+}
+
+/// Determines if an array contains all elements of a subset array in any order.
+/// - Parameters:
+///   - array: The array to search within
+///   - subset: The subset array to check for
+/// - Returns: `true` if all elements in `subset` are present in `array`, `false` otherwise
+internal func containsNonContiguousSubset<T: Equatable>(array: [T], subset: [T]) -> Bool {
+    for element in subset {
+        if !array.contains(element) {
+            return false
+        }
+    }
+    return true
```

Comment:
suggestion (non-blocking): for fun, you could also make this helper a one-line implementation! :D 

```
return subset.allSatisfy({ array.contains($0) })
```

Reply:
Thanks for this.  I'll address this in a follow-up change.